src/url.js

```diff
@@ -0,0 +1,27 @@
+import isUrl from 'is-url';
+
+import wrapValidator from './helpers/wrapValidator';
+
+function url(props, propName, componentName) {
+  const value = props[propName];
+
+  if (value == null || isUrl(value)) {
```

Comment:
it'd be better to follow the format of the other validators in this package, and have the optional one call into the required one after early-passing for `== null`

Reply:
oooo I was using [`nonNegativeNumber`](https://github.com/airbnb/prop-types/blob/master/src/nonNegativeNumber.js) as an implementation guide but I like this way better (like in [`integer`](https://github.com/airbnb/prop-types/blob/master/src/integer.js))!